src/appleseed-max-impl/appleseedinteractive/interactiverenderercontroller.cpp

```diff
@@ -39,6 +42,110 @@
 
 namespace asr = renderer;
 
+
+void UpdateObjectInstanceAction::update()
+{
+    // todo: handle objects with optimized_for_instancing flag on
+    renderer::Assembly* assembly = m_session->m_project->get_scene()->assemblies().get_by_name("assembly");
+
+    if (m_session->m_object_inst_map.count(wide_to_utf8(m_node->GetName())) > 0)
+    {
+        m_session->m_object_map.erase(m_node->GetObjectRef());
+        asr::ObjectInstance* object_instance = static_cast<asr::ObjectInstance*>(m_session->m_object_inst_map[wide_to_utf8(m_node->GetName())]);
```

Comment:
Why is a cast necessary here?

Reply:
it casts from `renderer::Entity*`, I thought it's necessary. `m_object_inst_map` is `std::map<std::string, renderer::Entity*>`